indico/modules/events/ical.py

```diff
@@ -12,72 +12,121 @@
 
 from indico.core import signals
 from indico.core.config import config
+from indico.core.db.sqlalchemy.protection import ProtectionMode
 from indico.util.date_time import now_utc
 from indico.util.signals import values_from_signal
 
 
-def event_to_ical(event, user=None):
+def generate_basic_component(entity, uid=None, url=None):
+    """Generate an iCalendar component with basic common properties.
+
+    :param entity: Event/session/contribution where properties come from
+    :param uid: UID for the component
+    :param url: URL for the component (defaults to `entity.external_url`)
+
+    :return: iCalendar event with basic properties
+    """
+    component = Event()
```

Comment:
I would prefer `icalendar.Event()` here to be more clear that it's not an indico `Event` instance